lib/realm-verifier/src/lib.rs

```diff
@@ -7,55 +7,99 @@ use tinyvec::ArrayVec;
 
 pub const MAX_MEASUREMENT_SIZE: usize = 64;
 
-pub type MeasurementValue = ArrayVec<[u8; MAX_MEASUREMENT_SIZE]>;
+#[derive(Debug, Clone, Copy)]
+pub struct MeasurementValue {
+    value: ArrayVec<[u8; MAX_MEASUREMENT_SIZE]>,
+}
+
+impl MeasurementValue {
+    pub fn init(len: usize) -> Self {
+        let mut av = ArrayVec::new();
+        av.resize(len, 0);
+        Self {
+            value: av
+        }
+    }
+
+    pub fn as_slice(&self) -> &[u8] {
+        self.value.as_slice()
+    }
+
+    pub fn as_mut_slice(&mut self) -> &mut[u8] {
+        self.value.as_mut_slice()
+    }
+}
+
+#[derive(Debug)]
+pub struct HashAlgo {
+    pub name: String,
+    pub len: usize,
+}
 
 #[derive(Debug)]
 pub struct RealmMeasurements {
     pub initial: MeasurementValue,
     pub extensible: Vec<[MeasurementValue; CLAIM_COUNT_REALM_EXTENSIBLE_MEASUREMENTS]>,
+    pub hash_algo: HashAlgo,
 }
 
 #[derive(Debug)]
 pub struct RealmVerifier {
     reference_measurements: RealmMeasurements,
 }
 
+macro_rules! eq_msg {
```

Comment:
Would be nicer to be a template (PartialEq) function. I don't think macro is needed here.